File: frontend/src/types/domain.ts
export type DetectionStatus = 'NORMAL' | 'UNKNOWN_ANOMALY' | 'KNOWN_ATTACK';
export type Severity = 'LOW' | 'MEDIUM' | 'HIGH' | 'CRITICAL';
export type ModelRole = 'SUPERVISED_CLASSIFIER' | 'STATISTICAL_ANOMALY_DETECTOR';
export type ModelProtocol = 'A' | 'B';
export type DeploymentTier = 'PRODUCTION_DEFAULT' | 'PRODUCTION_ALTERNATIVE' | 'RESEARCH_EVALUATION';

export type AttackFamily =
  | 'BENIGN'
  | 'BOTNET'
  | 'BRUTE_FORCE'
  | 'DDOS'
  | 'DOS'
  | 'HEARTBLEED'
  | 'INFILTRATION'
  | 'PORT_SCAN'
  | 'WEB_ATTACK';

File: backend/app/domain.py
"""Stable domain types shared by API and decision logic."""

from enum import StrEnum


class DetectionStatus(StrEnum):
    NORMAL = "NORMAL"
    KNOWN_ATTACK = "KNOWN_ATTACK"
    UNKNOWN_ANOMALY = "UNKNOWN_ANOMALY"


class Severity(StrEnum):
    LOW = "LOW"
    MEDIUM = "MEDIUM"
    HIGH = "HIGH"
    CRITICAL = "CRITICAL"
